### ISSUE TITLE
Only save the best (on validation set)

### ISSUE THREAD

**Issue by icebreaker2** [icebreaker2/AI#41]
* Save only those weights that are better on validation set
* Apply this to callbacks
* check that callbacks use validation value and not training data values for decision on betterness
